pkgload:::generate_db() fails

I'm trying the new compilation database functionality in the 1.4.0 release and it keeps failing with:

```
Warning message:
! System command 'make' failed 
```
 Digging deeper and running `pkgload:::generate_db()` directly I get the following, more useful, error

```
Error in `processx::run("make", c(makevars, "print-compilers"), env = env)`:
! System command 'make' failed
---
Exit status: 2
Stderr:
/usr/lib64/R/etc/Makeconf:10: /usr/lib64/R/share/make/vars.mk: No such file or directory
make: *** No rule to make target '/usr/lib64/R/share/make/vars.mk'.  Stop.
```

Note I'm running on R 4.4.1 on Fedora Linux 40. For me vars.mk is found in /usr/share/R/make/vars.mk not the address looked for above.

~Currently I'm unsure if this is pkgload / processx /R or Fedora related (tagging @Enchufa2 in case of the latter).~

--- a/R/compilation-db.R
+++ b/R/compilation-db.R
@@ -201,7 +201,7 @@ compilers <- function() {
     out <- processx::run("make", c(makevars, "print-compilers"), env = env)
   )
 
-  compilers <- strsplit(out$stdout, "\n")[[1]]
+  compilers <- strsplit(trimws(out$stdout), "\n")[[1]]
 
   # Remove arguments
   compilers <- strsplit(compilers, " ")

--- a/R/compilation-db.R
+++ b/R/compilation-db.R
@@ -194,7 +194,7 @@ compilers <- function() {
   env <- c(
     "current",
     R_INCLUDE_DIR = Sys.getenv("R_INCLUDE_DIR", unset = fs::path(R.home(), "include")),
-    R_SHARE_DIR = Sys.getenv("R_SHARE_DIR", fs::path(R.home(), "share"))
+    R_SHARE_DIR = Sys.getenv("R_SHARE_DIR", unset = fs::path(R.home(), "share"))
   )
 
   pkgbuild::with_build_tools(

--- a/R/compilation-db.R
+++ b/R/compilation-db.R
@@ -58,7 +58,7 @@ has_compilation_db <- function(desc) {
 
   out <- as.logical(field)
   check_bool(out, arg = "Config/build/compilation-database")
-  
+
   out
 }
 
@@ -193,8 +193,8 @@ compilers <- function() {
   # These variables are normally set by frontends but just in case
   env <- c(
     "current",
-    R_INCLUDE_DIR = fs::path(R.home(), "include"),
-    R_SHARE_DIR = fs::path(R.home(), "share")
+    R_INCLUDE_DIR = Sys.getenv("R_INCLUDE_DIR", unset = fs::path(R.home(), "include")),
+    R_SHARE_DIR = Sys.getenv("R_SHARE_DIR", fs::path(R.home(), "share"))
   )
 
   pkgbuild::with_build_tools(

